pkg/apis/azureprovider/v1alpha1/azuremachineproviderconfig_types.go

```diff
@@ -75,8 +75,8 @@ type AzureMachineProviderSpec struct {
 	Vnet string `json:"vnet"`
 
 	// Availability Zone for the virtual machine.
-	// If nil, a zone will be randomly chosen from the list of zones for the location.
-	// If the virtual machine should be deployed to no zone, it must be explicitly set to empty string.
+	// If nil, the virtual machine should be deployed to no zone
```

Comment:
What does "deploy to _no_ zone" mean/imply?

Reply:
That your resource is not tied to an specific zone, see: 
https://docs.microsoft.com/en-us/azure/virtual-machines/windows/regions-and-availability
https://docs.microsoft.com/en-us/azure/availability-zones/az-overview
This is just exposing the provider choices to consumers:
- nil - no zone
- "" random zone
- "1" explicit zone

Reply:
Nil doesn't imply no-zone, but it implies don't care and you can make the choice and therefore that the best place for random.

"" Implies explicitly no-zone

Reply:
We might even drop the "random" case. If consumer doesn't care,  they do nothing, no zone is set. I consumer cares it sets zone explicitly - There's no value on random choice and it does not guarantee any kind of tolerance to failures or benefit.
The installer can either ensure even zone spread of machineSets or no zoned machines
https://github.com/openshift/installer/pull/1949

Reply:
dropping the random zone code in favor of explicit sounds good to me.